src/util/codec/convert.rs

```diff
@@ -41,8 +39,17 @@ pub fn bytes_to_int(bytes: &[u8]) -> Result<i64> {
     Ok(r)
 }
 
-/// `bytes_to_f64` converts a byte array to a float64 in best effort.
-pub fn bytes_to_f64(bytes: &[u8]) -> Result<f64> {
+/// `bytes_to_int` converts a byte arrays to an i64 in best effort.
+/// TODO: handle overflow.
+pub fn bytes_to_int(ctx: &EvalContext, bytes: &[u8]) -> Result<i64> {
+    let s = try!(str::from_utf8(bytes)).trim();
+    let vs = get_valid_float_prefix(s);
```

Comment:
Just follows TiDB https://github.com/pingcap/tidb/blob/bb66fbe6b0ff7c09057d8b6821cec6c4832e2628/util/types/convert.go#L138.
However, overflow is not handled yet.